### PR TITLE
Add an issue template for bug reports

### LINKED ISSUE #2978: Add github issue template
This issue proposes adding a github issue template to help instruct users on filing bug reports, as per discussion with @diml at https://github.com/ocaml/dune/issues/2927#issuecomment-559503309 

I thought this could be achieved via a PR, but github now advises using their new issue template workflow. See the following for instructions on adding the template: https://help.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository#creating-issue-templates

Here's a rough proposal for the basic issue template, just to get things going. Please change as you see fit. (The following is elaborated from [stevemao's github-issue-templates repo](https://github.com/stevemao/github-issue-templates/blob/master/simple/ISSUE_TEMPLATE.md)).

```markdown
<!-- Thank you for filing an issue to help us improve Dune! -->
## Expected Behavior


## Actual Behavior


## Reproduction

<!-- If possible, please open a PR with a test that reproduces the problem.
      Adding tests is easy, see  https://github.com/ocaml/dune/blob/master/HACKING.md#tests -->

- PR with a reproducing test: 

<!-- Otherwise, please detail how to the behavior can be reproduced -->

1.
1.
1.

## Specifications

- Version (output of `dune --version`):
- Operating system (distribution and version):

<!-- remove the following if not used -->
## Additional information 

- Link to gist with verbose output (run with `--verbose` flag): 
```

### PR DESCRIPTION
**Pull request by shonfeder**
Closes #2978

It should still be possible for users to open blank issues.

### CODE REVIEW

**Comment by shonfeder** [comment]
(Force push was necessary to add the signoff line, sine I forget it while going through the github issue template configuration forms.)

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+## Specifications
+
+- Version (output of `dune --version`):
+- Operating system (distribution and version):
+
+<!-- remove the following if not used -->
+## Additional information 
+
+- Link to gist with verbose output (run with `--verbose` flag):
```
Comment (diml) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
- Link to gist with verbose output (run `dune` with `--verbose` flag):
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+---
+name: Bug report
+about: File an issue to help us improve
+title: ''
+labels: bug
```
Comment (diml) [inline .github/ISSUE_TEMPLATE/bug_report.md]: I'd leave this blank and the maintainers decide whether it is a bug or not.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+<!-- Otherwise, please detail how to the behavior can be reproduced -->
+
+1.
+1.
+1.
+
+## Specifications
+
+- Version (output of `dune --version`):
```
Comment (diml) [inline .github/ISSUE_TEMPLATE/bug_report.md]: the version of ocaml is often useful as well:

```suggestion
- Version of `dune` (output of `dune --version`):
- Version of `ocaml` (output of `ocamlc --version`):
```

**Comment by shonfeder** [comment]
Fixed. Thanks!

**Pull request merged**